src/GraphQL/Resolvers/EventStreamMethodResolver.cs

```diff
@@ -0,0 +1,86 @@
+using System.Linq.Expressions;
+using System.Reflection;
+using GraphQL.Subscription;
+
+namespace GraphQL.Resolvers
+{
+    /// <summary>
+    /// A precompiled event stream resolver for a specific <see cref="MethodInfo"/>.
+    /// Calls the specified method (with the specified arguments) and returns the value of the method.
+    /// </summary>
+    internal class EventStreamMethodResolver : MemberResolver, IAsyncEventStreamResolver
```

Comment:
This resolver implements both `IAsyncEventStreamResolver` and `IFieldResolver` and inherits much of its functionality from `MemberResolver`.  I did not attempt to implement `IEventStreamResolver`, for two reasons (1) a single `Task` allocation when setting up a new event stream is not so bad, (2) once we switch to `ValueTask`, there will only be one interface anyway.

Reply:
The reason this resolver implements `IFieldResolver` is because a typical event stream resolver returns an `IObservable<T>` and then when there is a data push from the event stream, GraphQL executes the `IFieldResolver` for the same field, with `T` as the root, which in turn is immediately returned as the source for the child graph.  For `AutoRegisteringObjectGraphType` this will be simplified so that any method that returns `IObservable<T>` will automatically set up a subscription field and handle this legwork.  If a user really wants the child graph to use a child property of the observable, they can call Select on the observable before returning it.  All this allows for proper type inference.

Long story short: it's prepwork for a subsequent PR and not needed quite yet.